modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2189,6 +2190,19 @@ else if (Objects.equals(
 			List<String> oDataFilterStrings = TransformUtil.transform(
 				(List<ObjectFilter>)objectFieldSettingsValues.get("filters"),
 				objectFilter -> {
+					if (StringUtil.equals(
+							objectFilter.getFilterType(),
+							ObjectFilterConstants.TYPE_CURRENT_USER)) {
+
+						objectFilter.setJSON(
+							String.valueOf(
+								_jsonFactory.createJSONObject(
+								).put(
+									"currentUserId",
+									PrincipalThreadLocal.getUserId()
+								)));
```

Comment:
@gabrielwas can we change to:

```
objectFilter.setJSON(
	JSONUtil.put(
		"currentUserId", PrincipalThreadLocal.getUserId()
	).toString()
)
```
or do you prefer to use the `_jsonFactory`?

Reply:
@carolmariaabb better your way! Thanks!